Formula/flake8.rb

```diff
@@ -37,26 +38,10 @@ class Flake8 < Formula
   end
 
   def install
-    venv = virtualenv_create(libexec, "python3")
-    resource("entrypoints").stage do
-      # Without removing this file, `pip` will ignore the `setup.py` file and
-      # attempt to download the [`flit`](https://github.com/takluyver/flit)
-      # build system.
-      rm_f "pyproject.toml"
-      venv.pip_install Pathname.pwd
-    end
-    (resources.map(&:name).to_set - ["entrypoints"]).each do |r|
-      venv.pip_install resource(r)
-    end
-    venv.pip_install_and_link buildpath
+    virtualenv_install_with_resources
   end
 
   test do
-    xy = Language::Python.major_minor_version "python3"
-    # flake8 version 3.7.8 will fail this test with `E203` warnings.
-    # Adding `E203` to the list of ignores makes the test pass.
-    # Remove the customized ignore list once the problem is fixed upstream.
-    system "#{bin}/flake8", "#{libexec}/lib/python#{xy}/site-packages/flake8",
-           "--ignore=E121,E123,E126,E226,E24,E704,W503,W504,E203"
+    system "#{bin}/flake8", "--exit-zero", "#{libexec}/lib/python*/site-packages/flake8"
```

Comment:
This makes the test useless since it'll never exit with a failure now.

Reply:
Thanks for pointing out. To be honest, I have a similar concern, here are some reasons and any comments are welcome:

- The test here means test for the installation of `flake8`. If flake8 is not installed correctly, this test will fail absolutely. And according to the help message of flake8 [1], `--exit-zero` means `Exit with status code "0" even if there are errors.` Note the `errors` here are **style errors** of the target, not the errors of flake8 installation. So I think it should be OK to use `--exit-zero` just like using error ignores previously.
- The previous error ignoring list is outdated. Only `E203` and `W504` is needed now from my test. To avoid similar situations, maybe it is better just use an universal `--exit-zero` flag just as `pylint` [2] does.
- I also tried to use test without `--exit-zero`, and `brew test flake8` still exit with code 0 even though the actual command `/usr/local/Cellar/flake8/3.7.9_1/bin/flake8 /usr/local/Cellar/flake8/3.7.9_1/libexec/lib/python*/site-packages/flake8` exit with code 1 as expected. I am not sure whether I triggered it correctly, but anyway, adding an extra `--exit-zero` does no further harm.

[1] https://flake8.pycqa.org/en/latest/user/invocation.html#invoking-flake8
[2] https://github.com/Homebrew/homebrew-core/blob/master/Formula/pylint.rb

Reply:
I would rather prefer a minimal python file (test.py) with 2-3 lines of python, that is then tested by flake 8. Something small, that will not break in the future. Testing the flake8 source code with flake8 is a good idea, but is likely to break in the case where upstream's code is not as clean as it should be.

Reply:
Hi @iMichka, so you mean to use something like [pylint](https://github.com/Homebrew/homebrew-core/blob/master/Formula/pylint.rb)?

I am OK with that but just to clarify, maybe `--exit-zero` is also a work around for the "clean" problem you mentioned.

Reply:
No, this does not involve pylint.

I was thinking about something like (untested):
```
(testpath/"test.py").write <<~EOS
import sys
a = 1
print("test")
EOS

system "#{bin}/flake8", "test.py"
````

Maybe assert the exit code too to check if it did run without error.

Reply:
OK, gotcha, will do soon.

Just another concern as mentioned previously. Seems `brew test flake8` will pass even the exit code is not 0. Just removing the `--exit-zero` then we can reproduce it. Is there anything wrong I made for this?